pkg/envtest/webhook.go

```diff
@@ -0,0 +1,355 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envtest
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	admissionreg "k8s.io/api/admissionregistration/v1beta1"
```

Comment:
what was that issue? I was talking with joe about proper support of v1 webhooks, I'll be creating an issue for that. But I assume this is something more subtle... :-)

Reply:
I was more suggesting we start with webhooks/v1 instead of v1beta1. If we don't, we might have to add support later in the future.

We had the same issue for `envtest` CRD, and to be backward compatible we have to handle both versions, which makes the whole code a little tricky. https://github.com/kubernetes-sigs/controller-runtime/pull/752

Reply:
oh I see, yeah that makes sense to me if @DirectXMan12 is OK with that...

Reply:
i agree with this. if there's an interface here too, can we introduce a helper that uses this interface? if not....ok then.

basically I want to abstract the version from the object as much as possible and generally select the client's storage version (as we would with CRDs).

generally :+1: on latest version selection from our level of abstraction, and binding CR release versions to certain core GVK versions.

Reply:
I think probably the safest method here is some sort of helper that can deal with v1 and v1beta1 webhooks like we do with CRDs

Reply:
since we're prob stuck with v1beta1 for at least a bit longer

Reply:
(basically should be able to copy the code from the CRD handling logic, more-or-less)